From 127328ecd3996a572d202d9a33da98cad24ecc75 Mon Sep 17 00:00:00 2001
From: Dominik Csapak <d.csapak@proxmox.com>
Date: Tue, 17 Jan 2017 17:24:03 +0100
Subject: [PATCH 06/10] change src directory for images/js files

since they will be in /novnc/
also change the directory in the build script

Signed-off-by: Dominik Csapak <d.csapak@proxmox.com>
---
 app/ui.js |  2 +-
 vnc.html  | 56 ++++++++++++++++++++++++++++----------------------------
 2 files changed, 29 insertions(+), 29 deletions(-)

diff --git a/app/ui.js b/app/ui.js
index 662352f..c523fcd 100644
--- a/app/ui.js
+++ b/app/ui.js
@@ -1720,7 +1720,7 @@ const UI = {
 const LINGUAS = ["de", "el", "es", "nl", "pl", "sv", "tr", "zh_CN", "zh_TW"];
 l10n.setup(LINGUAS);
 if (l10n.language !== "en" && l10n.dictionary === undefined) {
-    WebUtil.fetchJSON('app/locale/' + l10n.language + '.json', (translations) => {
+    WebUtil.fetchJSON('/novnc/app/locale/' + l10n.language + '.json', function (translations) {
         l10n.dictionary = translations;
 
         // wait for translations to load before loading the UI
diff --git a/vnc.html b/vnc.html
index bc1a47d..d17ee81 100644
--- a/vnc.html
+++ b/vnc.html
@@ -28,7 +28,7 @@
     <meta name="apple-mobile-web-app-capable" content="yes" />
     <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
     <!-- Stylesheets -->
-    <link rel="stylesheet" href="app/styles/base.css" />
+    <link rel="stylesheet" href="/novnc/app/styles/base.css" />
     <link rel="stylesheet" href="/novnc/app/styles/pve.css" />
 
     <!--
@@ -37,11 +37,11 @@
     -->
 
     <!-- this is included as a normal file in order to catch script-loading errors as well -->
-    <script type="text/javascript" src="app/error-handler.js"></script>
+    <script type="text/javascript" src="/novnc/app/error-handler.js"></script>
 
     <!-- begin scripts -->
     <!-- promise polyfills promises for IE11 -->
-    <script src="vendor/promise.js"></script>
+    <script src="/novnc/vendor/promise.js"></script>
     <!-- ES2015/ES6 modules polyfill -->
     <script type="module">
         window._noVNC_has_module_support = true;
@@ -50,12 +50,12 @@
         window.addEventListener("load", function() {
             if (window._noVNC_has_module_support) return;
             var loader = document.createElement("script");
-            loader.src = "vendor/browser-es-module-loader/dist/browser-es-module-loader.js";
+            loader.src = "/novnc/vendor/browser-es-module-loader/dist/browser-es-module-loader.js";
             document.head.appendChild(loader);
         });
     </script>
     <!-- actual script modules -->
-    <script type="module" crossorigin="anonymous" src="app/ui.js"></script>
+    <script type="module" crossorigin="anonymous" src="/novnc/app/ui.js"></script>
     <!-- end scripts -->
 </head>
 
@@ -80,49 +80,49 @@
             <h1 class="noVNC_logo" translate="no"><span>no</span><br />VNC</h1>
 
             <!-- Drag/Pan the viewport -->
-            <input type="image" alt="viewport drag" src="app/images/drag.svg"
+            <input type="image" alt="viewport drag" src="/novnc/app/images/drag.svg"
                 id="noVNC_view_drag_button" class="noVNC_button noVNC_hidden"
                 title="Move/Drag Viewport" />
 
             <!--noVNC Touch Device only buttons-->
             <div id="noVNC_mobile_buttons">
-                <input type="image" alt="No mousebutton" src="app/images/mouse_none.svg"
+                <input type="image" alt="No mousebutton" src="/novnc/app/images/mouse_none.svg"
                     id="noVNC_mouse_button0" class="noVNC_button"
                     title="Active Mouse Button"/>
-                <input type="image" alt="Left mousebutton" src="app/images/mouse_left.svg"
+                <input type="image" alt="Left mousebutton" src="/novnc/app/images/mouse_left.svg"
                     id="noVNC_mouse_button1" class="noVNC_button"
                     title="Active Mouse Button"/>
-                <input type="image" alt="Middle mousebutton" src="app/images/mouse_middle.svg"
+                <input type="image" alt="Middle mousebutton" src="/novnc/app/images/mouse_middle.svg"
                     id="noVNC_mouse_button2" class="noVNC_button"
                     title="Active Mouse Button"/>
-                <input type="image" alt="Right mousebutton" src="app/images/mouse_right.svg"
+                <input type="image" alt="Right mousebutton" src="/novnc/app/images/mouse_right.svg"
                     id="noVNC_mouse_button4" class="noVNC_button"
                     title="Active Mouse Button"/>
-                <input type="image" alt="Keyboard" src="app/images/keyboard.svg"
+                <input type="image" alt="Keyboard" src="/novnc/app/images/keyboard.svg"
                     id="noVNC_keyboard_button" class="noVNC_button"
                     value="Keyboard" title="Show Keyboard" />
             </div>
 
             <!-- Extra manual keys -->
             <div id="noVNC_extra_keys">
-                <input type="image" alt="Extra keys" src="app/images/toggleextrakeys.svg"
+                <input type="image" alt="Extra keys" src="/novnc/app/images/toggleextrakeys.svg"
                     id="noVNC_toggle_extra_keys_button" class="noVNC_button"
                     title="Show Extra Keys"/>
                 <div class="noVNC_vcenter">
                 <div id="noVNC_modifiers" class="noVNC_panel">
-                    <input type="image" alt="Ctrl" src="app/images/ctrl.svg"
+                    <input type="image" alt="Ctrl" src="/novnc/app/images/ctrl.svg"
                         id="noVNC_toggle_ctrl_button" class="noVNC_button"
                         title="Toggle Ctrl"/>
-                    <input type="image" alt="Alt" src="app/images/alt.svg"
+                    <input type="image" alt="Alt" src="/novnc/app/images/alt.svg"
                         id="noVNC_toggle_alt_button" class="noVNC_button"
                         title="Toggle Alt"/>
-                    <input type="image" alt="Tab" src="app/images/tab.svg"
+                    <input type="image" alt="Tab" src="/novnc/app/images/tab.svg"
                         id="noVNC_send_tab_button" class="noVNC_button"
                         title="Send Tab"/>
-                    <input type="image" alt="Esc" src="app/images/esc.svg"
+                    <input type="image" alt="Esc" src="/novnc/app/images/esc.svg"
                         id="noVNC_send_esc_button" class="noVNC_button"
                         title="Send Escape"/>
-                    <input type="image" alt="Ctrl+Alt+Del" src="app/images/ctrlaltdel.svg"
+                    <input type="image" alt="Ctrl+Alt+Del" src="/novnc/app/images/ctrlaltdel.svg"
                         id="noVNC_send_ctrl_alt_del_button" class="noVNC_button"
                         title="Send Ctrl-Alt-Del" />
                 </div>
@@ -130,13 +130,13 @@
             </div>
 
             <!-- Shutdown/Reboot -->
-            <input type="image" alt="Shutdown/Reboot" src="app/images/power.svg"
+            <input type="image" alt="Shutdown/Reboot" src="/novnc/app/images/power.svg"
                 id="noVNC_power_button" class="noVNC_button"
                 title="Shutdown/Reboot..." />
             <div class="noVNC_vcenter">
             <div id="noVNC_power" class="noVNC_panel">
                 <div class="noVNC_heading">
-                    <img src="app/images/power.svg"> Power
+                    <img src="/novnc/app/images/power.svg"> Power
                 </div>
                 <input type="button" id="noVNC_shutdown_button" value="Shutdown" />
                 <input type="button" id="noVNC_reboot_button" value="Reboot" />
@@ -145,13 +145,13 @@
             </div>
 
             <!-- Clipboard -->
-            <input type="image" alt="Clipboard" src="app/images/clipboard.svg"
+            <input type="image" alt="Clipboard" src="/novnc/app/images/clipboard.svg"
                 id="noVNC_clipboard_button" class="noVNC_button"
                 title="Clipboard" />
             <div class="noVNC_vcenter">
             <div id="noVNC_clipboard" class="noVNC_panel">
                 <div class="noVNC_heading">
-                    <img src="app/images/clipboard.svg"> Clipboard
+                    <img src="/novnc/app/images/clipboard.svg"> Clipboard
                 </div>
                 <textarea id="noVNC_clipboard_text" rows=5></textarea>
                 <br />
@@ -161,19 +161,19 @@
             </div>
 
             <!-- Toggle fullscreen -->
-            <input type="image" alt="Fullscreen" src="app/images/fullscreen.svg"
+            <input type="image" alt="Fullscreen" src="/novnc/app/images/fullscreen.svg"
                 id="noVNC_fullscreen_button" class="noVNC_button noVNC_hidden"
                 title="Fullscreen" />
 
             <!-- Settings -->
-            <input type="image" alt="Settings" src="app/images/settings.svg"
+            <input type="image" alt="Settings" src="/novnc/app/images/settings.svg"
                 id="noVNC_settings_button" class="noVNC_button"
                 title="Settings" />
             <div class="noVNC_vcenter">
             <div id="noVNC_settings" class="noVNC_panel">
                 <ul>
                     <li class="noVNC_heading">
-                        <img src="app/images/settings.svg"> Settings
+                        <img src="/novnc/app/images/settings.svg"> Settings
                     </li>
                     <li style="display:none;">
                         <label><input id="noVNC_setting_shared" type="checkbox" /> Shared Mode</label>
@@ -246,7 +246,7 @@
             </div>
 
             <!-- Connection Controls -->
-            <input type="image" alt="Disconnect" src="app/images/disconnect.svg"
+            <input type="image" alt="Disconnect" src="/novnc/app/images/disconnect.svg"
                 id="noVNC_disconnect_button" class="noVNC_button"
                 title="Disconnect" />
 
@@ -265,7 +265,7 @@
         <div id="noVNC_connect_dlg">
             <div class="noVNC_logo" translate="no"><span>no</span>VNC</div>
             <div id="noVNC_connect_button"><div>
-                <img src="app/images/connect.svg"> Connect
+                <img src="/novnc/app/images/connect.svg"> Connect
             </div></div>
         </div>
     </div>
@@ -306,8 +306,8 @@
     </div>
 
     <audio id="noVNC_bell">
-        <source src="app/sounds/bell.oga" type="audio/ogg">
-        <source src="app/sounds/bell.mp3" type="audio/mpeg">
+        <source src="/novnc/app/sounds/bell.oga" type="audio/ogg">
+        <source src="/novnc/app/sounds/bell.mp3" type="audio/mpeg">
     </audio>
  </body>
 </html>
-- 
2.11.0

